adr/recipes/raw_coverage.py

```diff
@@ -42,14 +42,83 @@ def run(args):
                         help="Ignore performance tests in aggregating and reporting jobs.")
     args = parser.parse_args(args)
 
-    def minimumJobs(args):
+    def artifactCount(args):
         query_args = vars(args)
-        result = run_query('raw_coverage', **query_args)
+        result = run_query('raw_coverage_count', **query_args)
+        for item in result['data']:
+            return item[0]
+
+    def buildDirList(dirs):
+        dirs.sort()
+        all_dirs = {}
+        last_dir = ''
+        for line in dirs:
+            l = line.strip()
+            if not l.endswith('/'):
+                l = "%s/" % l
+            if l == '/':
+                continue
+
+            args.path = l
+            count = int(artifactCount(args))
+            if count <= 0:
+                continue
+
+            if last_dir and l.startswith(last_dir) or l == last_dir:
+                continue
+
+            if l in all_dirs:
+                continue
+
+            # check if the parent is already in the dir list
+            if '/'.join(l.split('/')[0:-1]) in all_dirs:
+                continue
+
+            if count < 50000:
+                print("adding %s to all_dirs: %s" % (l, count))
```

Comment:
At the top of your file, add:

    log = logging.getLogger('adr')

Then turn these into:

    log.debug('adding ...')

This way, when running with `adr --quiet`, these messages won't get printed. This is important if we ever want to upload this to the gist.